src/step.rs

```diff
@@ -96,57 +100,73 @@ where
     F: Fn(&InterpCx) -> ShouldContinue,
 {
     let mut message = None;
-    loop {
-        if pcx.ecx.stack().len() <= 1 && is_ret(&pcx.ecx) {
```

Comment:
The new code is not equivalent to the old code. The old code would prevent stepping when doing so would exit the program. The new code does allow stepping out of the `main` function.

Reply:
Oh… Thanks for explaining what this piece of code is supposed to do! ☺